cherrypy/process/servers.py

```diff
@@ -185,14 +185,24 @@ def _get_base(self):
         if not self.httpserver:
             return ''
         host, port = self.bind_addr
-        if getattr(self.httpserver, 'ssl_adapter', None):
-            scheme = 'https'
-            if port != 443:
-                host += ':%s' % port
+        if sys.version_info[0] > 2:
```

Comment:
This condition checks whether the version of python is greater than 2 or not. Since the code failed temporarily failed the tests with certain python 3.x versions. The alternative code is executed after being checked by sys.version_info which works on python 3.x. The original code works fine with python 2.x